jsapp/js/components/library/singleCollectionStore.es6

```diff
@@ -0,0 +1,341 @@
+import _ from 'underscore';
+import Reflux from 'reflux';
+import {hashHistory} from 'react-router';
+import {
+  SEARCH_CONTEXTS,
+  searchBoxStore
+} from '../header/searchBoxStore';
+import assetUtils from 'js/assetUtils';
+import {
+  isOnSingleCollectionRoute,
+  getCurrentCollectionUID
+} from './libraryUtils';
+import {actions} from 'js/actions';
+import {
+  ORDER_DIRECTIONS,
+  ASSETS_TABLE_COLUMNS
+} from './assetsTable';
+
+// Mostly copied from `myLibraryStore`, so far only small changes made to get
+// collection assets to display in table
+const singleCollectionStore = Reflux.createStore({
+  /**
+   * A method for aborting current XHR fetch request.
+   * It doesn't need to be defined upfront, but I'm adding it here for clarity.
+   */
+  abortFetchData: undefined,
+  previousPath: null,
+  previousSearchPhrase: searchBoxStore.getSearchPhrase(),
+  PAGE_SIZE: 100,
+  DEFAULT_ORDER_COLUMN: ASSETS_TABLE_COLUMNS.get('date-modified'),
+
+  isVirgin: true,
+
+  data: {
+    isFetchingData: false,
+    currentPage: 0,
+    totalPages: null,
+    totalUserAssets: null,
+    totalSearchAssets: null,
+    assets: [],
+    metadata: {}
+  },
+
+  init() {
+    this.fetchDataDebounced = _.debounce(this.fetchData.bind(true), 2500);
+
+    this.setDefaultColumns();
+
+    hashHistory.listen(this.onRouteChange.bind(this));
+    searchBoxStore.listen(this.searchBoxStoreChanged);
+    actions.library.moveToCollection.completed.listen(this.onMoveToCollectionCompleted);
+    actions.library.subscribeToCollection.completed.listen(this.fetchData.bind(this, true));
+    actions.library.unsubscribeFromCollection.completed.listen(this.fetchData.bind(this, true));
+    actions.library.searchMyLibraryMetadata.completed.listen(this.onSearchMetadataCmpleted);
+    actions.resources.loadAsset.completed.listen(this.onAssetChanged);
+    actions.resources.updateAsset.completed.listen(this.onAssetChanged);
+    actions.resources.cloneAsset.completed.listen(this.onAssetCreated);
+    actions.resources.createResource.completed.listen(this.onAssetCreated);
+    actions.resources.deleteAsset.completed.listen(this.onDeleteAssetCompleted);
+    // These are the changed requests, not all have been updated yet. Trying to
+    // just get assets to show up first
+    actions.library.searchMyCollectionAssets.started.listen(this.onSearchStarted);
+    actions.library.searchMyCollectionAssets.completed.listen(this.onSearchCompleted);
+    actions.library.searchMyCollectionAssets.failed.listen(this.onSearchFailed);
+    // TODO Improve reaction to uploads being finished after task is done:
+    // https://github.com/kobotoolbox/kpi/issues/476
+    actions.resources.createImport.completed.listen(this.fetchDataDebounced);
```

Comment:
I'm not 100% sure, but I think single collection page doesn't need to react to imports, so we can delete this call (with comments) and delete `fetchDataDebounced` and delete `_` import.